tests/jest/web-push-notifications.js

```diff
@@ -0,0 +1,100 @@
+        makeChatManager(user)
+          .connect()
+          .then(user => {
+            return user.enablePushNotifications()
+          })
+          .then(() => {
+            return mockBeamsCalls
+          }),
+      user,
+    )
+    expect(mockBeamsCalls.startHasBeenCalled).toBeTruthy()
+    expect(mockBeamsCalls.stopHasBeenCalled).toEqual(false)
+    expect(mockBeamsCalls.setUserIdHasBeenCalled).toBeTruthy()
+    expect(mockBeamsCalls.setUserIdHasBeenCalledWithUserId).toEqual(user.id)
+    expect(mockBeamsCalls.setUserIdTokenProviderFetchedToken.token).toBeTruthy()
+  })
+
+  test("registers with Beams if `showNotificationsTabClosed` is set explicitly to true", async () => {
+    const user = await helpers.makeUser("default")
+    const mockBeamsCalls = await page.evaluate(
+      user =>
+        makeChatManager(user)
+          .connect()
+          .then(user => {
+            return user.enablePushNotifications({
+              showNotificationsTabClosed: true,
+            })
+          })
+          .then(() => {
+            return mockBeamsCalls
+          }),
+      user,
+    )
+    expect(mockBeamsCalls.startHasBeenCalled).toBeTruthy()
+    expect(mockBeamsCalls.stopHasBeenCalled).toEqual(false)
+    expect(mockBeamsCalls.setUserIdHasBeenCalled).toBeTruthy()
+    expect(mockBeamsCalls.setUserIdHasBeenCalledWithUserId).toEqual(user.id)
+    expect(mockBeamsCalls.setUserIdTokenProviderFetchedToken.token).toBeTruthy()
+  })
+
+  test("does NOT register with Beams if `showNotificationsTabClosed` is set to false", async () => {
+    const user = await helpers.makeUser("default")
+    const mockBeamsCalls = await page.evaluate(
+      user =>
+        makeChatManager(user)
+          .connect()
+          .then(user => {
+            return user.enablePushNotifications({
+              showNotificationsTabClosed: false,
+            })
+          })
+          .then(() => {
+            return mockBeamsCalls
+          }),
+      user,
+    )
+    expect(mockBeamsCalls.startHasBeenCalled).toEqual(false)
+    expect(mockBeamsCalls.stopHasBeenCalled).toBeTruthy()
+    expect(mockBeamsCalls.setUserIdHasBeenCalled).toEqual(false)
+    expect(mockBeamsCalls.setUserIdHasBeenCalledWithUserId).toBeNull()
+    expect(mockBeamsCalls.setUserIdTokenProviderFetchedToken).toEqual(false)
+  })
+
+  test("chat manager succeeds to disable Beams notifications", async () => {
+    const user = await helpers.makeUser("default")
+    const mockBeamsCalls = await page.evaluate(
+      user =>
+        makeChatManager(user)
+          .connect()
+          .then(user => {
+            user.enablePushNotifications({ showNotificationsTabClosed: false })
+            return user
+          })
+          .then(user => {
+            return user.disablePushNotifications()
+          })
+          .then(() => {
+            return mockBeamsCalls
+          }),
+      user,
+    )
+
+    expect(mockBeamsCalls.stopHasBeenCalled).toBeTruthy()
+  })
```

Comment:
We probably also want a test to ensure that tab open notifications have been disabled. The mocks only cover the tab closed case I believe.

Reply:
It's not immediately obvious to me how to mock out the tab open function calls. Testing the tab closed stuff is easier because the Beams function is injected. Ideas I have all involve injecting more stuff into ChatManager, which seems undesirable.  I could mock the whole ChatManager, but given these are basically integration tests that would also be weird. Open to suggestions!

Reply:
The approach I was taking in https://github.com/pusher/chatkit-client-js/pull/197 was to [inject a mock `Notification` constructor](https://github.com/pusher/chatkit-client-js/pull/197/files#diff-446b292502cd8afca5d7fa4e98e28d91R5). https://github.com/pusher/chatkit-client-js/pull/197/files#diff-ba2314b0298c25b349f0afbdcd9552bfR32

Maybe I can add a test for disabling notifications when I rebase that branch, so I'm happy to approve this without that test if you want!